apicast/src/provider.lua

```diff
@@ -372,16 +376,25 @@ function _M.access(service)
     params.app_id = parameters[credentials.app_id]
     params.app_key = parameters[credentials.app_key] -- or ""  -- Uncoment the first part if you want to allow not passing app_key
 
+    -- XXX: why do we need to include the app_key into the cached key? the app (and its access) are identified
```

Comment:
Answering my own question:
Using the correct `app_key` should result in storing `200` in cache for `service_id:app_id:app_key`, while when the invalid `app_key` is used with the correct `app_id`, the above cached value will not be used.